lib/AST/Decl.cpp

```diff
@@ -5103,8 +5103,9 @@ bool FuncDecl::isXCTestUnitTest() const {
   //    Objective-C reflection)...
   ASTContext &ctx = getASTContext();
   if (!ctx.LangOpts.EnableObjCInterop &&
-      (getFormalAccess() < Accessibility::Internal ||
-       parentClass->getFormalAccess() < Accessibility::Internal))
+      ((hasAccessibility() && getFormalAccess() < Accessibility::Internal) ||
+       (parentClass->hasAccessibility() &&
+        parentClass->getFormalAccess() < Accessibility::Internal)))
```

Comment:
Something is going horribly wrong on the Linux builders, which prompted this, I'll find the right thing to do here.